torch/csrc/jit/fuser/executor.cpp

```diff
@@ -297,9 +297,14 @@ bool runFusion(const int64_t key, Stack& stack) {
   auto& spec = *(*maybe_spec);
 
   // Acquires inputs from stack
-  auto inputs = fmap(last(stack, spec.nInputs()), [](const IValue& i) {
-    return i.toTensor();
-  });
+  auto all_inputs = last(stack, spec.nInputs());
+  std::vector<at::Tensor> inputs;
+  inputs.reserve(all_inputs.size());
+  for (auto& i : all_inputs) {
+    if (i.isTensor()) {
```

Comment:
Can't we simply have an invariant that tensor inputs precede other inputs? Then, you could compute the length of this loop once, and avoid unnecessary checks on size inputs.

Reply:
:+1:  We already have that invariant.